merlin-server/sql/merlin/tables/anchor_validation_status.sql

```diff
@@ -0,0 +1,341 @@
+  returns boolean
+  security definer
+  language plpgsql as $$
+  declare
+    net_offset interval;
+    _anchor_id integer;
+    _start_offset interval;
+    _anchored_to_start boolean;
+  begin
+    select anchor_id, start_offset, anchored_to_start
+    from activity_directive
+    where (id, plan_id) = (_activity_id, _plan_id)
+    into _anchor_id, _start_offset, _anchored_to_start;
+
+    if(_start_offset < '0'  and _anchored_to_start) then -- only need to check if anchored to start or something with a negative offset
+      with recursive anchors(plan_id, activity_id, anchor_id, anchored_to_start, start_offset, total_offset) as (
+          select _plan_id, _activity_id, _anchor_id, _anchored_to_start, _start_offset, _start_offset
+        union
+          select ad.plan_id, ad.id, ad.anchor_id, ad.anchored_to_start, ad.start_offset, anchors.total_offset + ad.start_offset
+          from activity_directive ad, anchors
+          where anchors.anchor_id is not null                               -- stop at plan
+            and (ad.plan_id, ad.id) = (anchors.plan_id, anchors.anchor_id)
+            and anchors.anchored_to_start                                  -- or, stop at end-time offset
+      )
+      select total_offset  -- get the id of the activity that the selected activity is anchored to
+      from anchors a
+      where a.anchor_id is null
+        and a.anchored_to_start
+      limit 1
+      into net_offset;
+
+      if(net_offset < '0') then
+        raise notice 'Activity Directive % has a net negative offset relative to Plan Start.', _activity_id;
+
+        insert into anchor_validation_status (activity_id, plan_id, reason_invalid)
+        values (_activity_id, _plan_id, 'Activity Directive ' || _activity_id || ' has a net negative offset relative to Plan Start.')
+        on conflict (activity_id, plan_id) do update
+          set reason_invalid = 'Activity Directive ' || excluded.activity_id || ' has a net negative offset relative to Plan Start.';
+        return true;
+     end if;
+    end if;
+
+    return false;
+    end
+  $$;
+comment on function is_net_negative_plan_start(_activity_id integer, _plan_id integer) is e''
+  'Returns true if the specified activity has a net negative offset from plan start. Otherwise, returns false.\n'
+  'If true, writes to anchor_validation_status.';
+
+/*
+  When an activity directive is anchored to another activity directive's end time, it must simplify to a positive offset,
+  as simulation can't handle a negative offset (since simulation won't know its start time until the anchoring activity has finished,
+  which will always place the anchored activity's start time in the past)
+
+  Throws an exception if:
+    - The activity is anchored to itself
+    - A cycle is detected
+  For other erroneous states that can wait until simulation time to be resolved, it writes to 'anchor_validation_status's 'reason_invalid' field and then returns.
```

Comment:
wording suggestion - feel free to wordsmith further. I found the existing wording a bit unclear
```suggestion
  For other erroneous states that we allow to exist, as long as they're resolved before simulation time, it writes to 'anchor_validation_status's 'reason_invalid' field and then returns.
```

Reply:
How about:

```
For all other invalid states, it writes to 'anchor_validation_status's 'reason_invalid' field and then returns.
If the activity's anchor is valid, then the 'reason_invalid' field on the activity's entry in 'anchor_validation_status' is set to ''.
```

Reply:
Sounds great to me :+1: